wikipedia_connector/wiki.py

```diff
@@ -99,48 +82,77 @@ def get_clean_text(self, text):
         text = re.sub(r'[^\x00-\x7F]+',' ', text)
         return text
 
-try:
-    if flag_0 == "dataframe":
-        input_list = []
-        list2 = []
-        disambiguation = 0
-        if(topics.shape[0]>0):
-            for i in range(topics.shape[0]):
-                abc1 = WikiPage(str(topics.iloc[i, 0]))
-                abc11 = abc1.get_wiki_page(str(topics.iloc[i, 0]))[0]
-                input_list.append(abc11)
-                if(abc1.get_wiki_page(str(topics.iloc[i, 0]))[1] == '1'):
-                    disambiguation = disambiguation + 1
-                list2.append(str(topics[i]))
+
+def wiki_main():
+    parser = argparse.ArgumentParser(description="""Preprocessor""")
+    parser.add_argument('-f','--topics', action='store', dest='topics', default="table,chalk,starcraft,predator", required=True, help="""wikipedia topics""")
+    parser.add_argument('--project_dir', action='store', dest='project_dir',
+                        help="""--- For inner use of cnvrg.io ---""")
+    parser.add_argument('--output_dir', action='store', dest='output_dir',
+                        help="""--- For inner use of cnvrg.io ---""")
+    parser.add_argument(        "--local_dir",
+        action="store",
+        dest="local_dir",
+        required=False,
+        default=cnvrg_workdir,
+        help="""--- The path to save the dataset file to ---""",)
+    
+    args = parser.parse_args()
+    topic = args.topics
+    file_dir = args.local_dir
+    if topic.endswith(".csv"):
+        topics = pd.read_csv(topic)
+        flag_0 = "dataframe"
+    else:
+        topics = topic.split(",")
+        flag_0 = "list"
+
+
+    try:
+        if flag_0 == "dataframe":
```

Comment:
Here is a marker for test case. 
Condition 1 flag_0 == dataframe

Reply:
@orionsBeltWest This condition is now checked